### ISSUE TITLE
Missing 'Referenced by' links for macros (C/C++)

### ISSUE THREAD

**Issue by HeikoAtWork** [doxygen/doxygen#10343]
**Describe the bug**
In the HTML documentation generated by Doxygen 1.9.8 from C/C++ source code the 'Referenced by' links for macros (#define ...) are missing if REFERENCED_BY_RELATION = YES is set in the configuration. These links have been there in the documentation generated by older Doxygen versions (e.g. Doxygen 1.6.2).

**Expected behavior**
If REFERENCED_BY_RELATION = YES is set in the Doxygen configuration (Doxyfile) then 'Referenced by' links should be generated also for C/C++ macros (#define ...) in the HTML output.

**Screenshots**
Doxygen 1.9.8: Macro documentation without referenced-by links
<img width="165" alt="Doxygen 1 9 8 macro documentation without referenced-by links" src="https://github.com/doxygen/doxygen/assets/146977167/a7f340a8-e0b6-4473-9758-d400cd9355e8">

Doxygen 1.6.2: Macro documentation with referenced-by links
<img width="284" alt="Doxygen 1 6 2 macro documentation with referenced-by links" src="https://github.com/doxygen/doxygen/assets/146977167/b990abe0-fec1-41ce-a814-048d6e971bff">

**Version**
Doxygen version 1.9.8 on Windows 10, 64 bit.

**Comment by HeikoAtWork** [comment]
[HelloWorldExampleWithMacro.zip](https://github.com/doxygen/doxygen/files/12814102/HelloWorldExampleWithMacro.zip)

Here is a simple example explaining the issue (including C code, Doxyfile and HTML output of Doxygen 1.6.2 and 1.8.9). For the HELLO_TEXT macro therein the old Doxygen 1.6.2 generates a "Referenced by ..." link. This link however is missing in the HTML output generated by Doxygen 1.9.8.

**Comment by albert-github** [comment]
Looks like that the information vanished between 1.8.14 and 1.8.15, have to do some bisecting...
(I did try yesterday something as well but apparently didn't send the comment but only set the labels).

**Comment by albert-github** [comment]
Did the bisecting and the result is:
```
0697535ad38ed122964c4673b102a8e30ad4369f is the first bad commit
commit 0697535ad38ed122964c4673b102a8e30ad4369f
Date:   Wed Oct 17 19:10:24 2018 +0200

    Add commands to handle referenced by relation and references relation

    Analogous to call graph and caller graph this patch provides an implementation for the referenced by relation and references relation.
    Providing the commands:
    - referencedbyrelation
    - hidereferencedbyrelation
    - referencesrelation
    - hidereferencesrelation

    Motivation is that some lists can get extremely large and also there is now more symmetry between the textual and graphical out.
```

Probably an extra call to a setting routine for macros is necessary, implementation was done analogous to call / caller graphs but macros are never shown here (and consequently probably the related call to referenced by relation is missing).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10346

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
